src/v/cluster/topics_frontend.cc

```diff
@@ -640,6 +640,11 @@ ss::future<std::error_code> topics_frontend::finish_moving_partition_replicas(
         return ss::make_ready_future<std::error_code>(
           errc::no_leader_controller);
     }
+    // optimization: if update is not in progress return early
+    if (!_topics.local().is_update_in_progress(ntp)) {
+        return ss::make_ready_future<std::error_code>(
```

Comment:
When we've cancelled and hit this path, what prevents the reconcile loop getting stuck (i.e. calling repeatedly) on the process_partition_update for this partition?

Reply:
cancelling partition movement is finished with `update_finished` command